qcodes/instrument_drivers/oxford/IPS120.py

```diff
@@ -19,32 +19,42 @@
 log = logging.getLogger(__name__)
 
 class OxfordInstruments_IPS120(VisaInstrument):
-    """This is the python driver for the Oxford Instruments IPS 120 Magnet Power Supply
+    """This is the driver for the Oxford Instruments IPS 120 Magnet Power Supply
 
-    Usage:
-    Initialize with:
-    magnet = qcodes.instrument_drivers.oxford.IPS120.OxfordInstruments_IPS120(name='magnet', address='ASRL4::INSTR')
-
-    Note: Since the ISOBUS allows for several instruments to be managed in parallel, the command
-    which is sent to the device starts with '@n', where n is the ISOBUS instrument number.
+    The IPS 120 can connect through both RS232 serial as well as GPIB. The
+    commands sent in both cases are similar. When using the serial connection,
+    commands are prefaced with '@n' where n is the ISOBUS number.
     """
 
-    def __init__(self, name, address, number=2, **kwargs):
+    _STATUS_MODE2 = {0: "At rest",
+            1: "Sweeping",
+            2: "Sweep limiting",
+            3: "Sweeping & sweep limiting"}
+
+    _STATUS_SWITCH_HEATER = {
+            0: "Off magnet at zero (switch closed)",
+            1: "On (switch open)",
+            2: "Off magnet at field (switch closed)",
+            5: "Heater fault (heater is on but current is low)",
+            8: "No switch fitted"}
+
+
+    def __init__(self, name, address, use_gpib=False, number=2, **kwargs):
         """Initializes the Oxford Instruments IPS 120 Magnet Power Supply.
 
         Args:
             name (string)    : name of the instrument
             address (string) : instrument address
-            number (int)     : ISOBUS instrument number
+            use_gpib (bool)  : whether to use GPIB or serial
```

Comment:
I am not that familiar with the IPS120. Are GPIB and serial the only 2 supported ways of communicating. The Mercury also supports comm over ethernet which is similar to the gpib (so that the serial) is the odd one out. But I am not sure if that applies to this one too?

Reply:
Only these two.

Reply:
Cool then this is fine